src/header/common/link.rs

```diff
@@ -0,0 +1,975 @@
+            Ok(RelationType::Section)
+        } else if "self".eq_ignore_ascii_case(s) {
+            Ok(RelationType::RelationTypeSelf)
+        } else if "service".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Service)
+        } else if "start".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Start)
+        } else if "stylesheet".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Stylesheet)
+        } else if "subsection".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Subsection)
+        } else if "successor-version".eq_ignore_ascii_case(s) {
+            Ok(RelationType::SuccessorVersion)
+        } else if "up".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Up)
+        } else if "version-history".eq_ignore_ascii_case(s) {
+            Ok(RelationType::VersionHistory)
+        } else if "via".eq_ignore_ascii_case(s) {
+            Ok(RelationType::Via)
+        } else if "working-copy".eq_ignore_ascii_case(s) {
+            Ok(RelationType::WorkingCopy)
+        } else if "working-copy-of".eq_ignore_ascii_case(s) {
+            Ok(RelationType::WorkingCopyOf)
+        } else {
+            match Uri::new(s) {
+                Err(_) => Err(::Error::Header),
+                Ok(uri) => Ok(RelationType::ExtRelType(uri)),
+            }
+        }
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Utilities
+////////////////////////////////////////////////////////////////////////////////
+
+struct SplitAsciiUnquoted<'a> {
+    src: &'a str,
+    pos: usize,
+    len: usize,
+    del: &'a str
+}
+
+impl<'a> SplitAsciiUnquoted<'a> {
+    fn new(s: &'a str, d: &'a str) -> SplitAsciiUnquoted<'a> {
+        SplitAsciiUnquoted{
+            src: s,
+            pos: 0,
+            len: s.len(),
+            del: d,
+        }
+    }
+}
+
+impl<'a> Iterator for SplitAsciiUnquoted<'a> {
+    type Item = &'a str;
+
+    fn next(&mut self) -> Option<&'a str> {
+        if self.pos < self.len {
+            let prev_pos = self.pos;
+            let mut pos = self.pos;
+
+            let mut in_quotes = false;
+
+            for c in self.src[prev_pos..].as_bytes().iter() {
+                in_quotes ^= *c == b'"';
+
+                if !in_quotes && self.del.as_bytes().contains(c) {
+                    break;
+                }
+
+                pos += 1;
+            }
+
+            self.pos = pos + 1;
+
+            Some(&self.src[prev_pos..pos])
+        } else {
+            None
+        }
+    }
+}
+
+fn from_str_delimited<T: FromStr>(s: &str, d: char) -> ::Result<Vec<T>> {
+    let mut v: Vec<T> = Vec::new();
+
+    for i in s.split(d) {
+        match T::from_str(i.trim()) {
+            Err(_) => return Err(::Error::Header),
+            Ok(t) => v.push(t),
+        }
+    }
+
+    Ok(v)
+}
+
+fn fmt_delimited<T: fmt::Display>(f: &mut fmt::Formatter, p: &[T], d: &str, s: &str, e: &str) -> fmt::Result {
```

Comment:
This function as well as `from_str_delimited` kind of overlap with the ones you have in `header/parsing.rs`. We could move them but I think it's fine like this.